src/invidious/user/imports.cr

```diff
@@ -310,7 +310,9 @@ struct Invidious::User
 
             db = DB.open("sqlite3://" + tempfile.path)
 
-            user.watched += db.query_all("SELECT url FROM streams", as: String)
+            user.watched += db.query_all(
+              "SELECT s.url FROM streams s JOIN stream_history sh ON s.uid = sh.stream_id",
```

Comment:
```suggestion
              "SELECT s.url FROM streams s JOIN stream_history sh ON s.uid = sh.stream_id ORDER BY sh.access_date asc",
```

Ordering by access date will also fix the order of the watch history in the event of repeated watches